File: server-side/lms/src/main/java/com/totfd/lms/service/SkillService.java
package com.totfd.lms.service;

import com.totfd.lms.dto.skill.request.SkillRequestDTO;
import com.totfd.lms.dto.skill.response.SkillResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface SkillService {
    SkillResponseDTO createSkill(SkillRequestDTO dto);
    SkillResponseDTO getSkillById(Long id);
    List<SkillResponseDTO> getAllSkills();
    Page<SkillResponseDTO> getAllSkillsWithPage(Pageable pageable);
    SkillResponseDTO updateSkill(Long id, SkillRequestDTO dto);
    void deleteSkill(Long id);
    SkillResponseDTO getSkillDetailsById(Long id);

}

File: client-side/client-side/src/services/skillService.js
import axiosInstance from "../api/axiosInstance";

export const createSkill = (data) => axiosInstance.post('/skills/addSkill', data);

export const getSkillById = (id) => axiosInstance.get(`/skills/${id}`);

export const getAllSkills = () => axiosInstance.get('/skills');

export const getPagedSkills = (page = 0, size = 20) =>
  axiosInstance.get(`/skills/page?page=${page}&size=${size}`);

export const updateSkill = (id, data) => axiosInstance.put(`/skills/${id}`, data);

export const deleteSkill = (id) => axiosInstance.delete(`/skills/${id}`);

export const getSkillWithTopics = (id) => axiosInstance.get(`/skills/skills/${id}/topics`);


